isok-cli/src/main.rs

```diff
@@ -0,0 +1,62 @@
+use core::str;
+
+use std::io;
+
+mod private_key;
+
+mod new_agent;
+use self::new_agent::NewAgentCommand;
+
+mod new_broker;
+use self::new_broker::NewBrokerCommand;
+
+// CLI
+
+/// Command line interface for creating `isok` brokers and agents.
+#[derive(Debug, Clone, clap::Parser)]
+#[clap(version, author)]
+pub struct Cli {
+    #[clap(subcommand)]
+    pub command: Command,
+}
+
+impl Cli {
+    pub fn from_env() -> Self {
+        match <Self as clap::Parser>::try_parse() {
+            Ok(cli) => cli,
+            Err(error) => error.exit(),
+        }
+    }
+
+    pub fn run(self) -> io::Result<()> {
+        self.command.run(&mut io::stdout().lock())
+    }
+}
+
+// COMMAND
+
+/// Subcommands for the [`Cli`].
+#[derive(Debug, Clone, clap::Subcommand)]
+pub enum Command {
+    NewBroker(NewBrokerCommand),
+    NewAgent(NewAgentCommand),
+}
+
+impl Command {
+    pub fn run(self, w: &mut impl io::Write) -> io::Result<()> {
+        match self {
+            Self::NewBroker(command) => command.run(w),
+            Self::NewAgent(command) => command.run(w),
+        }
+    }
+}
+
+#[test]
+fn verify_cli() {
+    use clap::CommandFactory;
+    Cli::command().debug_assert();
+}
+
+fn main() -> io::Result<()> {
```

Comment:
In CLIs, I started to use `eyre`. It helps to have a nice error for the user, with the stack of error. If you use eyre, you can replace the `io::Error` by `eyre::Report`. Doesn't apply for the rest of the CLI though. 

```
Error: Unable to load config file

Caused by:
    Config path provided is invalid

Location:
    isok-agent/src/main.rs:69:54
```